pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -232,6 +244,52 @@ func (ku *KubeUtil) getLocalPodList(ctx context.Context) (*PodList, error) {
 	return &pods, nil
 }
 
+// addContainerResourcesData modifies the given pod list, populating the
+// resources field of each container. If the pod resources API is not available,
+// this is a no-op.
+func (ku *KubeUtil) addContainerResourcesData(ctx context.Context, pods []*Pod) error {
+	if ku.podResources == nil {
+		return nil
+	}
+
+	containerToDevicesMap, err := ku.podResources.GetContainerToDevicesMap(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting container resources data: %w", err)
+	}
+
+	for _, pod := range pods {
+		ku.addResourcesToContainerList(containerToDevicesMap, pod, pod.Status.InitContainers)
+		ku.addResourcesToContainerList(containerToDevicesMap, pod, pod.Status.Containers)
+	}
+
+	return nil
+}
+
+func (ku *KubeUtil) addResourcesToContainerList(containerToDevicesMap map[ContainerKey][]*podresourcesv1.ContainerDevices, pod *Pod, containers []ContainerStatus) {
+	for i := range containers {
+		container := &containers[i] // take the pointer so that we can modify the original
```

Comment:
nit: I don't think you need this since containers[i] is a reference to the underlying array item already